tests/models/test_svc.py

```diff
@@ -0,0 +1,121 @@
+from numpy.testing import assert_allclose, assert_equal
+from sklearn.svm import SVC as SkSVC
+from scratchml.models.svc import SVC
+from scratchml.scalers import StandardScaler
+from ..utils import generate_classification_dataset, repeat
+import unittest
+import numpy as np
+import math
+
+
+class Test_SVC(unittest.TestCase):
+    """
+    Unit test class for the custom SVC implementation.
+    """
+
+    @repeat(3)
+    def test_binary_classification(self):
+        """
+        Test binary classification and compare the custom SVC with Scikit-Learn's SVC.
+        """
+        X, y = generate_classification_dataset(
+            n_samples=2000, n_features=4, n_classes=2
+        )
+
+        # Initialize and train both models
+        custom_svc = SVC(kernel="linear")
+        sklearn_svc = SkSVC(kernel="linear", max_iter=1000)
+
+        custom_svc.fit(X, y)
+        sklearn_svc.fit(X, y)
+
+        # Predict and score
+        custom_pred = custom_svc.predict(X)
+        sklearn_pred = sklearn_svc.predict(X)
+
+        custom_score = custom_svc.score(X, y)
+        sklearn_score = sklearn_svc.score(X, y)
+
+        # Assertions for binary classification
+        atol = math.floor(y.shape[0] * 0.1)
+        assert_equal(sklearn_svc.classes_, custom_svc.classes_)
+        assert_allclose(sklearn_pred, custom_pred, atol=atol)
+        assert abs(sklearn_score - custom_score) / abs(sklearn_score) < 0.05
+
+    @repeat(3)
+    def test_multi_class_classification(self):
+        """
+        Test multi-class classification and compare the custom SVC with Scikit-Learn's SVC.
+        """
+        # Use scaled data for both models
+        X, y = generate_classification_dataset(
+            n_samples=2000, n_features=4, n_classes=2
+        )
+        scaler = StandardScaler()
+        X = scaler.fit_transform(X)
+
+        # Initialize and train both models with adjusted max_iter and tol
+        custom_svc = SVC(kernel="linear", max_iter=1000, tol=1e-5)
+        sklearn_svc = SkSVC(kernel="linear", max_iter=1000, tol=1e-5)
+
+        custom_svc.fit(X, y)
+        sklearn_svc.fit(X, y)
+
+        custom_pred = custom_svc.predict(X)
+        sklearn_pred = sklearn_svc.predict(X)
+
+        custom_score = custom_svc.score(X, y)
+        sklearn_score = sklearn_svc.score(X, y)
+
+        atol = math.floor(y.shape[0] * 0.1)
+        assert_equal(sklearn_svc.classes_, custom_svc.classes_)
+        assert_allclose(sklearn_pred, custom_pred, atol=atol)
+        assert abs(sklearn_score - custom_score) / abs(sklearn_score) < 0.05
+
+    @repeat(3)
+    def test_rbf_kernel(self):
```

Comment:
It would be nice to add a few more validations, such as asserting the output type and shape of both implementations are equal, other parameters value, and so on. Check out other implementation tests to see if you get any idea to increment this.